src/main/scala/graphql/schema/CandidateModeType.scala

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package basic.graphql.schema
+
+import basic.ObservingMode
+import basic.search.Constraints
+import cats.effect._
+import sangria.schema._
+
+object CandidateModeType {
+
+  def apply[F[_]: Effect]: UnionType[Unit] =
+    UnionType(
+      name        = "CandidateModeType",
+      description = Some("Candidate observing mode."),
+      types       = List(spectroscopy[F])
+    )
+
+  def spectroscopy[F[_]: Effect]: ObjectType[Unit, (Constraints.Spectroscopy, ObservingMode.Spectroscopy)] =
+    ObjectType(
+      name  = "SpectroscopyObservingMode",
+      fields = fields(
+
+        Field(
+          name        = "instrument",
+          fieldType   = StringType,
+          resolve     = _.value._2.instrument.tag
```

Comment:
Yes, it's about passing back a value for this field that GraphQL knows how to turn into JSON. In this case a `String`.

In more detail, we're defining an

```scala
ObjectType[Unit, (Constraints.Spectroscopy, ObservingMode.Spectroscopy)]
```

Which means  *"here's metadata which describes, among other things, how to pull field values out of a `(Constraints.Spectroscopy, ObservingMode.Spectroscopy)` given a `Unit` as an arbitrary helper value (since we're not using it)."* In the database-backed Sangria example I showed you, this helper object is a database access object.

We're passing `Field` objects to the `ObjectType` constructor. The `Field` constructor has some inferred types. `resolve`'s type here is 

```scala
Context[Unit, (Constraints.Spectroscopy, ObservingMode.Spectroscopy)] => String
```

Where 

```scala
trait Context[C, V] {
  def ctx: C
  def value: V
  ...
}
```

and the `String` return type can be anything but it has to conform with `StringType` which is metadata for scalar strings and has type `OutputType[String]`.

So `_.value` here is the `(Constraints.Spectroscopy, ObservingMode.Spectroscopy)` that we're pulling values out of,